pkg/workloadmeta/collectors/internal/remote/process_collector/process_collector_test.go

```diff
@@ -231,11 +231,11 @@ func TestCollection(t *testing.T) {
 			errorResponse: true,
 		},
 	}
+	mockConfig := config.Mock(t)
+	mockConfig.Set("workloadmeta.remote_process_collector.enabled", true)
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			mockConfig := config.Mock(t)
```

Comment:
I was playing around with this locally, and this didn't fix the problem for me...

What did (as far as I can tell) is:

```
			ctx, cancel := context.WithCancel(context.TODO())
			err = collector.Start(ctx, mockStore)
			require.NoError(t, err)

			// Wait for gRPC calls to be sent
			time.Sleep(1 * time.Second)

			cancel()
```

instead of
```
			// Start collection
			err = collector.Start(context.TODO(), mockStore)
			require.NoError(t, err)
			// Wait for gRPC calls to be sent
			time.Sleep(1 * time.Second)
``` 
starting on line 274

Reply:
Or, well, a combination of both fixed it, this change is still needed it looks like

Reply:
Thanks @sblumenthal! I'm surprised by what could cause a data race here

Reply:
I think it has to do with the fact that we are running the server in a goroutine, and we were not explicitly stopping the server. Calling cancel here forces the loop [running here](https://github.com/DataDog/datadog-agent/blob/eba5f5c106c27c30b5c7b4124e224307b5a9a355/pkg/workloadmeta/collectors/internal/remote/generic.go#L172) to exit

Reply:
Indeed, it makes sense! Thanks for your help there.